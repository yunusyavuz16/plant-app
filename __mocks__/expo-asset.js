module.exports = {
  Asset: {
    loadAsync: jest.fn(),
  },
};