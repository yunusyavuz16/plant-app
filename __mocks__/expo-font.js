module.exports = {
  useFonts: () => [true, null],
};