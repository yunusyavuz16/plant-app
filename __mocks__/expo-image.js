const { View } = require("react-native");

module.exports = {
  Image: (props) => <View testID="mock-image" {...props} />,
};