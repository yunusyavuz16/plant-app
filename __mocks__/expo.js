module.exports = {
  __esModule: true,
  default: {
    registerRootComponent: jest.fn(),
  },
};