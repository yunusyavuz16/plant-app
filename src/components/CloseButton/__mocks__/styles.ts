export const mockStyles = {
  closeButton: {
    width: 32,
    height: 32,
    borderRadius: 16,
    backgroundColor: '#000000',
    alignItems: 'center',
    justifyContent: 'center',
    position: 'absolute',
    top: 17,
    right: 17,
    zIndex: 10,
  },
  closeText: {
    color: '#FFFFFF',
    fontSize: 24,
    textAlign: 'center',
    fontWeight: '400',
  },
};

export const mockTheme = {
  colors: {
    overlay: {
      dark: '#000000',
    },
    text: {
      white: '#FFFFFF',
    },
  },
  layout: {
    icon: {
      sm: 32,
    },
  },
  spacing: {
    xl: 16,
    xxl: 24,
  },
  typography: {
    size: {
      xl: 24,
    },
  },
};