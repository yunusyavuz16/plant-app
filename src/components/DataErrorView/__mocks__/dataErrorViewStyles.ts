export const mockTheme = {
  colors: {
    background: {
      white: '#FFFFFF',
      lightGray: '#F5F5F5',
    },
    text: {
      primary: '#000000',
      secondary: '#666666',
    },
  },
  spacing: {
    md: 12,
    lg: 16,
    xl: 24,
  },
  typography: {
    fonts: {
      Rubik: 'Rubik',
      RubikMedium: 'Rubik-Medium',
    },
    size: {
      sm: 14,
      md: 16,
      lg: 18,
    },
  },
};

export const mockStyles = {
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    padding: 24,
    backgroundColor: '#FFFFFF',
  },
  content: {
    alignItems: 'center',
    maxWidth: 300,
  },
  iconContainer: {
    width: 64,
    height: 64,
    borderRadius: 32,
    backgroundColor: '#F5F5F5',
    justifyContent: 'center',
    alignItems: 'center',
    marginBottom: 16,
  },
  icon: {
    fontSize: 32,
  },
  message: {
    fontFamily: 'Rubik-Medium',
    fontSize: 18,
    color: '#000000',
    textAlign: 'center',
    marginBottom: 12,
  },
  details: {
    fontFamily: 'Rubik',
    fontSize: 14,
    color: '#666666',
    textAlign: 'center',
    marginBottom: 16,
  },
  retryButton: {
    backgroundColor: '#000000',
    paddingHorizontal: 24,
    paddingVertical: 12,
    borderRadius: 12,
    marginTop: 12,
  },
  retryText: {
    fontFamily: 'Rubik-Medium',
    fontSize: 16,
    color: '#FFFFFF',
  },
};