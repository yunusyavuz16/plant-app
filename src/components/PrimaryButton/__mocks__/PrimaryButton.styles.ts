const styles = {
  container: {
    width: '100%',
    backgroundColor: '#4CAF50',
    borderRadius: 8,
    justifyContent: 'center',
    alignItems: 'center',
    height: 48,
  },
  label: {
    color: '#FFFFFF',
    fontSize: 16,
    fontFamily: 'SFProText-Bold',
    letterSpacing: 0.5,
  },
};

export default styles;