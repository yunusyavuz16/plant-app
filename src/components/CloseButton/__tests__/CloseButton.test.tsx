import React from 'react';
import { Text, TouchableOpacity } from 'react-native';
import renderer, { act } from 'react-test-renderer';
import { mockStyles, mockTheme } from '../__mocks__/styles';

// Mock the theme constants
jest.mock('@/constants/theme', () => mockTheme);

// Mock the safe area context
jest.mock('react-native-safe-area-context', () => ({
  useSafeAreaInsets: () => ({
    top: 47,
    right: 0,
    bottom: 34,
    left: 0,
  }),
}));

// Mock the styles
jest.mock('../CloseButton.styles', () => ({
  styles: mockStyles,
  getTopStyle: (insets: any) => ({
    top: insets.top + mockTheme.spacing.xxl,
  }),
}));

// Import the component after all mocks are set up
const CloseButton = require('../CloseButton').default;

describe('CloseButton', () => {
  const defaultProps = {
    handleClose: jest.fn(),
  };

  const renderComponent = (props = defaultProps) => {
    let component: renderer.ReactTestRenderer;
    act(() => {
      component = renderer.create(<CloseButton {...props} />);
    });
    return component!;
  };

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('renders correctly', () => {
    const tree = renderComponent().toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with correct structure', () => {
    const component = renderComponent();
    const touchable = component.root.findByType(TouchableOpacity);
    const text = component.root.findByType(Text);

    expect(touchable).toBeTruthy();
    expect(text).toBeTruthy();
    expect(text.props.children).toBe('Ã—');
  });

  it('applies correct styles', () => {
    const component = renderComponent();
    const touchable = component.root.findByType(TouchableOpacity);
    const text = component.root.findByType(Text);

    // Check button styles
    expect(touchable.props.style).toEqual([
      expect.objectContaining({
        width: 32,
        height: 32,
        borderRadius: 16,
        backgroundColor: '#000000',
        alignItems: 'center',
        justifyContent: 'center',
        position: 'absolute',
        top: 17,
        right: 17,
        zIndex: 10,
      }),
      expect.objectContaining({
        top: 71, // Mocked insets.top (47) + spacing.xxl (24)
      }),
    ]);

    // Check text styles
    expect(text.props.style).toEqual(expect.objectContaining({
      color: '#FFFFFF',
      fontSize: 24,
      textAlign: 'center',
      fontWeight: '400',
    }));
  });

  it('calls handleClose when pressed', () => {
    const mockHandleClose = jest.fn();
    const component = renderComponent({ handleClose: mockHandleClose });
    const touchable = component.root.findByType(TouchableOpacity);

    act(() => {
      touchable.props.onPress();
    });

    expect(mockHandleClose).toHaveBeenCalledTimes(1);
  });

  it('uses safe area insets correctly', () => {
    const component = renderComponent();
    const touchable = component.root.findByType(TouchableOpacity);

    // Check if the top style is calculated correctly using the mocked insets
    expect(touchable.props.style[1]).toEqual({
      top: 71, // Mocked insets.top (47) + spacing.xxl (24)
    });
  });

  it('handles multiple presses correctly', () => {
    const mockHandleClose = jest.fn();
    const component = renderComponent({ handleClose: mockHandleClose });
    const touchable = component.root.findByType(TouchableOpacity);

    // Simulate multiple presses
    act(() => {
      touchable.props.onPress();
      touchable.props.onPress();
      touchable.props.onPress();
    });

    expect(mockHandleClose).toHaveBeenCalledTimes(3);
  });

  it('handles undefined handleClose prop gracefully', () => {
    // This test ensures the component doesn't crash if handleClose is undefined
    expect(() => {
      renderComponent({ handleClose: undefined as any });
    }).not.toThrow();
  });
});