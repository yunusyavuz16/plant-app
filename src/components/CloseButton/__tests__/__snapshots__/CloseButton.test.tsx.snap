// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CloseButton renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#000000",
      "borderRadius": 16,
      "height": 32,
      "justifyContent": "center",
      "opacity": 1,
      "position": "absolute",
      "right": 17,
      "top": 71,
      "width": 32,
      "zIndex": 10,
    }
  }
>
  <Text
    style={
      {
        "color": "#FFFFFF",
        "fontSize": 24,
        "fontWeight": "400",
        "textAlign": "center",
      }
    }
  >
    Ã—
  </Text>
</View>
`;
